plugins/Diagnostics/Diagnostic/RequiredPhpSetting.php

```diff
@@ -101,7 +101,7 @@ public function __toString()
         return $this->setting . ' ' . implode(' OR ', $checks);
     }
 
-    public function jsonSerialize()
+    public function jsonSerialize(): mixed
```

Comment:
In that case we can't be implementing `\JsonSerializable` since that's the cause of the warning.

Reply:
It can be suppressed using the `#[\ReturnTypeWillChange]` comment

Reply:
Yep, that was the alternative solution. Will update the PR to add that instead of the return type.

Reply:
PR updated

Reply:
@michalkleiner I just had another thought on that. Actually we could also simply set `string` as return type hint.
As this is more specific as `mixed`, it should be compatible with that definition, but should be a lot better than the `ReturnTypeWillChange` workaround.

Reply:
Yep, that works for me. Updated the PR to that effect. Do we want to add it to the `__toString` method as well since that's the direct source of the output for the method in question?

Reply:
Yes. why not.
Imho we should try to add return types where ever it is safely possible.